app/react/src/client/preview/render.tsx

```diff
@@ -28,9 +39,54 @@ export const render: ArgsStoryFn<ReactFramework> = (args, context) => {
 
 const renderElement = async (node: ReactElement, el: Element) =>
   new Promise((resolve) => {
-    ReactDOM.render(node, el, () => resolve(null));
+    // Create Root Element conditionally for new React 18 Root Api
+    const root = getReactRoot(el);
+
+    if (root) {
+      root.render(node);
+      setTimeout(() => {
+        resolve(null);
+      }, 0);
+    } else {
+      ReactDOM.render(node, el, () => resolve(null));
+    }
   });
 
+const unmountElement = (el: Element) => {
+  const root = nodes.get(el);
+  if (root && FRAMEWORK_OPTIONS?.newRootApi) {
+    root.unmount();
+    nodes.delete(el);
+  } else {
+    ReactDOM.unmountComponentAtNode(el);
+  }
+};
+
+const canUseReactRoot =
+  gte(reactDomVersion, '18.0.0') || coerce(reactDomVersion)?.version === '18.0.0';
+
+const getReactRoot = (el: Element): IRoot | null => {
+  if (!FRAMEWORK_OPTIONS?.newRootApi) {
+    return null;
```

Comment:
```js
import { once } from '@storybook/client-logger';

if(canUseReactRoot) {
  once.info('To use the new React18 API in Storybook, set reactOptions.newRootApi: true` in your main.js https://..../MIGRATION.md#react18-support')
}
```

Reply:
I'll take care of this one

Reply:
Unrelated observer so let me know if this is the improper place but..... 

Are you sure the config prop should be called `newRootApi`? I ask, because when later versions of React comes, this will be the standard api from my understanding, which means eventually this prop will need to be removed or switched to support old api as React depreciates the older way. 

From a less-obstructive-upgrade-path mentality, shouldn't it be something like:
- Check react version
- If < 18, assume the old way
- else, have the config actually be "oldRootApi" and allow using the old way if its configured, otherwise use the new way by default?

That way, there's an understanding that when you switch to 18 with storybook, you get the new api and if people need the old api until they get things upgraded they can enable it as such.  

Otherwise, I see down the line either cutting people off cold turkey when react no longer supports the old api, or having to transition people from newRootApi to oldRootApi anyway which would require another config change down the road.

Just a thought if it's worth considered.

Reply:
@valentinpalkovic we discussed @brutus-law 's  "opt out" approach as a possibility, but I think you had some reservations. Can you summarize them here so we can get more feedback? Thanks! 🙏

Reply:
I had just some reservations of not having a flag at all. Instead of having a opt-in flag, we can rather opt-out by using a flag like „oldRootApi“. Makes totally sense!